modules/dxp/apps/saml/saml-opensaml-integration/src/main/java/com/liferay/saml/opensaml/integration/internal/resolver/DefaultUserResolver.java

```diff
@@ -90,23 +90,28 @@ public User resolveUser(
 
 		User user = null;
 
-		long companyId = CompanyThreadLocal.getCompanyId();
-
 		String subjectNameIdentifier = getSubjectNameIdentifier(
 			userResolverSAMLContext);
 
+		SamlSpIdpConnection samlSpIdpConnection =
+			_samlSpIdpConnectionLocalService.getSamlSpIdpConnection(
+				CompanyThreadLocal.getCompanyId(),
+				userResolverSAMLContext.resolvePeerEntityId());
+
 		String subjectNameIdentifierType = getSubjectNameIdentifierType(
-			userResolverSAMLContext);
+			userResolverSAMLContext, samlSpIdpConnection.getNameIdFormat());
 
 		if (_samlProviderConfigurationHelper.isLDAPImportEnabled()) {
 			user = importLdapUser(
-				companyId, subjectNameIdentifier, subjectNameIdentifierType);
+				samlSpIdpConnection.getCompanyId(), subjectNameIdentifier,
```

Comment:
@stian-sigvartsen shoudl we also use `CompanyThreadLocal` here as well? I know it should not differ but I would try to stick to one source of truth just in case... WDYT?

Reply:
@csierra hmm, that would mean also passing companyId into addUser() because this is now sourcing it from SamlSpIdpConnection also.  I wanted to avoid that because of the confusion it might cause when reading the method impls, because then there would be 2 ways to obtain companyId. Maybe we could add a comparison of the 2 sources in resolveUser() to give confidence?

Reply:
I see... I guess the comparison is the best we can do... in case it was different it would mean the DB is somehow corrupted and we could stop with an error

Reply:
Upon further thought and offline conversation I'm convinced that you are right. Just in case in the future there are global scope `SamlSpIdpConnections` with companyId=0 retrieved with `CompanyThreadLocal.getCompanyId()` .